site/releases/2024-jan-22/highlights.qmd

```diff
@@ -0,0 +1,218 @@
+---
+title: "January 22, 2024"
+keywords: "release notes, ai risk, model risk management, ValidMind"
+---
+
+## Release highlights
+
+This release includes numerous improvements to the ValidMind Developer Framework, including new features for model and dataset initialization, easier testing, support for additional inputs and the Azure OpenAI API, updated notebooks, bug fixes, deprecations, and much more.
+
+### ValidMind Developer Framework (v1.25.3)
+
+<!---[SC-2628] `init_model` improvements  by @AnilSorathiya in [#304](https://github.com/validmind/developer-framework/pull/304)--->
+
+#### Improvements to `init_model`
+
+When initializing a model, you can now pass a dataset with pre-computed model predictions to improve performance. By default, if no prediction column is specified when calling `init_model`, the ValidMind Developer Framework will compute the model predictions as before.
```

Comment:
this description is a bit unclear to me. 
What does "compute model predictions as before" refer to?
What is meant by "improve performance"?

Reply:
I think these would mainly be questions for @AnilSorathiya since it's from his [#304](https://github.com/validmind/developer-framework/pull/304). Anil, would you be able to provide some answers to Mehdi's questions, please?

Reply:
> What is meant by "improve performance"?

We can reword this to communicate that users can save time when they have big datasets and (probably) have pre-computed model predictions somewhere.

```
When initializing a model, you can now pass a dataset with pre-computed model predictions if they are available.
```

> What does "compute model predictions as before" refer to?

Every time we call `init_model()` we run predictions on the entire dataset.

Suggestion: just remove the word "before":

```
By default, if no prediction column is specified when calling `init_model`, the ValidMind Developer Framework will compute the model predictions on the entire dataset.
```

Reply:
> this description is a bit unclear to me.
> What does "compute model predictions as before" refer to?
> What is meant by "improve performance"?

Currently, we run prediction on given datasets (training, test datasets) while calling `init_model` method and store prediction values in the model object (vm_model).  Therefore it takes time to create vm_model object. 
Now this functionality provides flexibility to pass `prediction values`  as a additional column in the datasets while creating vm_model object. If prediction column is provided, we don't need to run model prediction while creating `vm_mdoel`.  This means it's going to speed up.

There is another reason to do this as well. Sometime model developer wants to compute prediction while creating a model for just once. The reason could be datasets are very large. In this case he might perform prediction out side of VM env. 

If prediction column is not provided, it will follow current behaviour.

Reply:
Thank you both, @AnilSorathiya and @cachafla! I have updated the text as suggested by Andres. (If you would like me to include more of Anil's added explanation let me know. I found it helpful to understand the feature better but the gist seems to be conveived by Andres' rewording.)